quickstartKotlin/src/main/java/com/twilio/video/quickstart/kotlin/VideoActivity.kt

```diff
@@ -526,20 +555,23 @@ class VideoActivity : AppCompatActivity() {
     override fun onCreateOptionsMenu(menu: Menu): Boolean {
         val inflater = menuInflater
         inflater.inflate(R.menu.menu, menu)
+        turnSpeakerOnMenuItem = menu.findItem(R.id.menu_turn_speaker_on)
+        turnSpeakerOffMenuItem = menu.findItem(R.id.menu_turn_speaker_off)
         return true
     }
 
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         when (item.itemId) {
             R.id.menu_settings -> startActivity(Intent(this, SettingsActivity::class.java))
-            R.id.speaker_menu_item -> if (audioManager.isSpeakerphoneOn) {
-                audioManager.isSpeakerphoneOn = false
-                item.setIcon(R.drawable.ic_phonelink_ring_white_24dp)
-                isSpeakerPhoneEnabled = false
-            } else {
-                audioManager.isSpeakerphoneOn = true
-                item.setIcon(R.drawable.ic_volume_up_white_24dp)
-                isSpeakerPhoneEnabled = true
+            R.id.menu_turn_speaker_on, R.id.menu_turn_speaker_off -> {
```

Comment:
The logic here is a little bit confusing when looking at it for the first time. I recommend having two separate cases instead of having a combined case so you don't have to think about the boolean logic.